online-shop/src/app/login/login.component.ts

```diff
@@ -0,0 +1,30 @@
+import { Component} from '@angular/core';
+import { Credentials } from '../core/models/credentials.model';
+import { Store } from '@ngrx/store';
+import { Observable } from 'rxjs';
+import { AppState, selectAuthState } from '../store/app.states';
+import { LogIn } from '../store/actions/user.actions';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html'
+})
+export class LoginComponent {
+  credentials : Credentials = new Credentials();
+  state$: Observable<any>;
```

Comment:
You should try to define a type for this (based on the selector return type) instead of `any`.